backend/pkg/modules/internal/bfla/init.go

```diff
@@ -0,0 +1,541 @@
+	return p, nil
+}
+
+type eventAlerter struct {
+	accessor core.BackendAccessor
+}
+
+func (e eventAlerter) SetEventAlert(ctx context.Context, modName string, eventID uint, severity core.AlertSeverity) (err error) {
+	switch severity {
+	case core.AlertInfo:
+		err = e.accessor.CreateAPIEventAnnotations(ctx, modName, eventID, core.AlertInfoAnn)
+	case core.AlertWarn:
+		err = e.accessor.CreateAPIEventAnnotations(ctx, modName, eventID, core.AlertWarnAnn)
+	default:
+		return fmt.Errorf("unexpected severity")
+	}
+	if err != nil {
+		return fmt.Errorf("error createing an alert: %w", err)
+	}
+	return nil
+}
+
+func (p *bfla) EventNotify(ctx context.Context, event *core.Event) {
+	if err := p.eventNotify(ctx, event); err != nil {
+		log.Errorf("EventNotify: %s", err)
+	}
+}
+
+func (p *bfla) eventNotify(ctx context.Context, event *core.Event) (err error) {
+	log.Infof("[BFLA] received a new event for API(%v) Event(%v) ", event.APIEvent.APIInfoID, event.APIEvent.ID)
+	cmpTrace := &bfladetector.CompositeTrace{Event: event}
+	if cmpTrace.K8SDestination, cmpTrace.K8SSource, cmpTrace.DetectedUser, err = getBFLAAnnotations(ctx, p.accessor, event.APIEvent.ID); err != nil {
+		log.Errorf("unable to get bfla annotations: %s", err)
+	}
+	if err := p.addK8sDestination(ctx, cmpTrace, event.Telemetry, event.APIEvent.ID); err != nil {
+		log.Errorf("unable to add k8s destination: %s", err)
+	}
+	if err := p.addK8sSource(ctx, cmpTrace, event.Telemetry, event.APIEvent.ID); err != nil {
+		log.Errorf("unable to add k8s source: %s", err)
+	}
+	if err := p.addDetectedUser(ctx, cmpTrace, event.Telemetry, event.APIEvent.ID); err != nil {
+		log.Errorf("unable to add detected user: %s", err)
+	}
+
+	// using a go routine to send traces in order not to block other modules.
+	p.bflaDetector.SendTrace(cmpTrace)
+	return err
+}
+
+func (p *bfla) addDetectedUser(ctx context.Context, cmpTrace *bfladetector.CompositeTrace, trace *pluginsmodels.Telemetry, eventID uint) (err error) {
+	if cmpTrace.DetectedUser != nil {
+		return nil
+	}
+	if trace == nil {
+		return nil
+	}
+
+	cmpTrace.DetectedUser, err = bfladetector.GetUserID(convertHeadersToMap(trace.Request.Common.Headers))
+	if err != nil {
+		log.Error(err)
+	}
+	if cmpTrace.DetectedUser == nil {
+		return nil
+	}
+	cmpTrace.DetectedUser.IPAddress = trace.SourceAddress
+	annDest := core.Annotation{Name: bfladetector.DetectedIDAnnotationName}
+	if annDest.Annotation, err = json.Marshal(cmpTrace.DetectedUser); err != nil {
+		return fmt.Errorf("unable to marshal user: %w", err)
+	}
+	if err := p.accessor.CreateAPIEventAnnotations(ctx, bfladetector.ModuleName, eventID, annDest); err != nil {
+		return fmt.Errorf("failed to create event annotation: %w", err)
+	}
+	return nil
+}
+
+func convertHeadersToMap(headers []*pluginsmodels.Header) map[string]string {
+	hmap := map[string]string{}
+	for _, h := range headers {
+		hmap[h.Key] = h.Value
```

Comment:
The key should be lowercased, and all reference to constant headers name should be used lowercased.